### ISSUE TITLE
Add matrix generators and test suites for SpMV

### ISSUE THREAD

**Issue by willow-ahrens** [SparseRooflineBenchmark/SparseRooflineBenchmark#2]
Currently, we have a SuiteSparse [1] and RMAT [2,3] generator.

I think we need to consider adding a suite of 20 or so matrices that we consider "important" enough to warrant extensive study, as well as a few much larger suites of matrices to test interesting behaviors or properties. Since Spmv has been studied before, we can start with existing lists of benchmarks [5, 6, 7, 8].

[1] T. A. Davis and Y. Hu, “The university of Florida sparse matrix collection,” ACM Transactions on Mathematical Software, vol. 38, no. 1, pp. 1–25, Nov. 2011, doi: [10.1145/2049662.2049663](https://doi.org/10.1145/2049662.2049663).
[2] D. Chakrabarti, Y. Zhan, and C. Faloutsos, “R-MAT: A Recursive Model for Graph Mining,” in Proceedings of the 2004 SIAM International Conference on Data Mining (SDM), in Proceedings. , Society for Industrial and Applied Mathematics, 2004, pp. 442–446. doi: [10.1137/1.9781611972740.43](https://doi.org/10.1137/1.9781611972740.43).
[3] L. Hübschle-Schneider and P. Sanders, “Linear work generation of R-MAT graphs,” Network Science, vol. 8, no. 4, pp. 543–550, Dec. 2020, doi: [10.1017/nws.2020.21](https://doi.org/10.1017/nws.2020.21).
[4] D. Langr and P. Tvrdík, “Evaluation Criteria for Sparse Matrix Storage Formats,” IEEE Transactions on Parallel and Distributed Systems, vol. 27, no. 2, pp. 428–440, Feb. 2016, doi: [10.1109/TPDS.2015.2401575](https://doi.org/10.1109/TPDS.2015.2401575).
[5] P. Stathis, S. Vassiliadis, and S. Cotofana, “D-SAB: A Sparse Matrix Benchmark Suite,” in Parallel Computing Technologies, V. E. Malyshkin, Ed., in Lecture Notes in Computer Science. Berlin, Heidelberg: Springer, 2003, pp. 549–554. doi: [10.1007/978-3-540-45145-7_52](https://doi.org/10.1007/978-3-540-45145-7_52).
[6] S. Beamer, K. Asanović, and D. Patterson, “The GAP Benchmark Suite.” arXiv, May 16, 2017. Accessed: Jun. 13, 2023. [Online]. Available: http://arxiv.org/abs/1508.03619
[7] “Graph 500 | large-scale benchmarks.” Accessed: Jul. 22, 2022. [Online]. Available: https://graph500.org/
[8] R. Vuduc, J. W. Demmel, K. A. Yelick, S. Kamil, R. Nishtala, and B. Lee, “Performance Optimizations and Bounds for Sparse Matrix-Vector Multiply,” in SC ’02: Proceedings of the 2002 ACM/IEEE Conference on Supercomputing, Nov. 2002, pp. 26–26. doi: [10.1109/SC.2002.10025](https://doi.org/10.1109/SC.2002.10025).

**Comment by weiya711** [comment]
@willow-ahrens I know we already have SuiteSparse, but is it worth creating an issue to try and figure out which SuiteSparse matrices are most likely to be used with the SpMV expression (i.e. filtering out SuiteSparse matrices not used in SpMV problems)? This goes along with the idea of having "important" matrices. Since anyone can add matrix data to SuiteSparse, presumably many of the matrices are used for solver expressions or SpMM and not SpMV

**Comment by weiya711** [comment]
I'm also going to create a separate issue to add in the uniform distribution data to SpMV (since that is more bite sized)

**Comment by weiya711** [comment]
Also another question for @willow-ahrens, how did you come up with the number "20" for the set of important matrices? But I do agree that we should decompose this issue into another one about studying and selecting an "important" set of SpMV matrices
